### PR TITLE
format: fix issues with colors support

### LINKED ISSUE #2011: formatter: colorsEnabled: true not honoured
### 👓 What did you see?

When outputting to a non-TTY environment, using `colorsEnabled: true` in the format options doesn't result in colours being output.

### ✅ What did you expect to see?

Colours to be output, because of this logic:

```ts
if (doesNotHaveValue(configuration.options.colorsEnabled)) {
      typeOptions.parsedArgvOptions.colorsEnabled = (
        stream as TtyWriteStream
      ).isTTY
    }
```

In other words, if we specify the option, it should output colours without checking if the stream is TTY.

### 📦 Which tool/library version are you using?

cucumber-js 8.1.2

### 🔬 How could we reproduce it?

TBA

### PR DESCRIPTION
**Pull request by davidjgoss**
### 🤔 What's changed?

We recently switched from `colors` to `chalk` for handling terminal colors. `chalk` includes its own rather comprehensive support checks at runtime, which had the unintended effect of disregarding a user-provided `colorsEnabled: true` (fixes #2011).

Now, rather than our own support check (which was just `stream.isTTY`), we explicitly delegate to chalk's detection library, for the specific output stream in question, and then construct our own chalk instance with the result (having overrided it based on the format option if necessary). See the feature file and docs for more detail, but the upshot is:

- Auto detection of colors support is better than it was
- The `colorsEnabled` option now works again
- Users can also use `FORCE_COLOR` which chalk supports, and we recommend it (for reasons in the docs)

### 🏷️ What kind of change is this?

- :bank: Refactoring/debt/DX (improvement to code design, tooling, documentation etc. without changing behaviour)
- :bug: Bug fix (non-breaking change which fixes a defect)

### ♻️ Anything particular you want feedback on?

🚨 After considering this, I'd like to go a bit further and deprecate (with intent to eventually remove) the `colorsEnabled` format option in favour of having users just use the kinda-standard `FORCE_COLOR` environment variable, which can at least influence other tools like assertion libraries which use `chalk` directly (see #1551) and would remove some complexity on our side (as we'd just fully delegate the logic to `supports-color`). Mocha folks have also leant into `FORCE_COLOR` to some degree in https://github.com/mochajs/mocha/issues/3641

Unless there are some implications I'm not seeing for integrators using Cucumber programmatically - paging @nicojs @jan-molak?

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [x] I've changed the behaviour of the code
  - [x] I have added/updated tests to cover my changes.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.
- [x] Users should know about my change
  - [x] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

File: src/formatter/get_color_fns.ts [hunk 1]
```diff
@@ -48,3 +57,15 @@ export default function getColorFns(enabled: boolean): IColorFns {
...
   }
 }
+
+function detectSupport(
+  stream: Writable,
+  env: NodeJS.ProcessEnv,
+  enabled?: boolean
+): ColorInfo {
+  const support: ColorInfo = supportsColor(stream)
+  if ('FORCE_COLOR' in env || doesNotHaveValue(enabled)) {
```
Comment (davidjgoss) [inline src/formatter/get_color_fns.ts:67]: If `FORCE_COLOR` is present, we'll just let the library handle that and not bother with our own option.

File: src/types/supports-color/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+declare module 'supports-color' {
```
Comment (davidjgoss) [inline src/types/supports-color/index.d.ts:1]: DefinitelyTyped types for this module were wrong.

File: docs/formatters.md [hunk 3]
```diff
@@ -39,9 +39,17 @@ This option is repeatable, so you can use it multiple times and the objects will
 
 Some options offered by built-in formatters:
 
-- `colorsEnabled` - if set to `false`, colors in terminal output are disabled
+- `colorsEnabled` - see below
```
Comment (aurelien-reeves) [inline docs/formatters.md:42]:
```suggestion
- `colorsEnabled` - [see below](#colored-output)
```

File: src/formatter/get_color_fns.ts (hunk 1 above)
Comment (aurelien-reeves) [reply src/formatter/get_color_fns.ts:67]: Maybe that comment would be useful in the code itself?

**Review by aurelien-reeves — COMMENTED** [review]
Looks good 🙂

Changelog entry is missing

Also, I do not get why do we add a stream to `getColorFns`?

**Comment by davidjgoss** [comment]
> Also, I do not get why do we add a stream to `getColorFns`?

We pass in the stream that the formatter is outputting to, so that `supports-colors` will detect the colors support for that particular stream, otherwise it would just look at `process.stdout` which may or may not be right.

**Review by aurelien-reeves — APPROVED** [review]

**Review by aurelien-reeves — APPROVED** [review]

**Comment by binarymist** [comment]
Thanks Team, tested and all good!

**Pull request merged**